catalog/ui/catalog-ui-search/src/main/webapp/decorator/menu-navigation.decorator.js

```diff
@@ -13,52 +13,82 @@
  *
  **/
 /*global define*/
+var $ = require('jquery');
+
+const expandComposedMenus = (menuItems) => {
+    let expandedItems = [];
+    let expanded = false;
+    menuItems.forEach(element => {
+        if ($(element).hasClass('composed-menu')) {
+            expanded = true;
+            expandedItems = expandedItems.concat($(element).children().toArray());
+        } else {
+            expandedItems.push(element);
+        }
+    });
+    if (expanded === false) {
+        return expandedItems;
+    } else {
+        return expandComposedMenus(expandedItems);
+    }
+};
+
+const handleArrowKey = (componentView, up) => {
+    let menuItems = componentView.handleTabIndexes();
+    let currentActive = menuItems.filter(element => $(element).hasClass('is-active'))[0];
+    var potentialNext = menuItems[menuItems.indexOf(currentActive) + (up === true ? -1 : 1)];
```

Comment:
It's typically best to stick with `===` in javascript in order to avoid unintended side effects from type coersion, things like empty strings or empty arrays or even the number 0 turning out to be false.

Reply:
I get that, but this `=== true` is not stopping type coercion; it's just making it more confusing. If an empty string or 0 or other falsy values are passed in here, they will still be treated as false because there is no check for `=== false`. Type coercion is still affecting this statement; all that's changed with `=== true` is that truthy values besides `true` will now be treated like `false`, which is unintuitive.